src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -980,10 +980,15 @@ void Transformations::MainSnippets(void) {
         }
         return true;
     };
-    auto is_unsupported_parallel_work_amount = [&](const std::shared_ptr<const ov::Node>& n, const ov::Shape& shape) {
-        const size_t parallel_work_amount = std::accumulate(shape.rbegin() + 2, shape.rend(), 1, std::multiplies<size_t>());
+    auto is_unsupported_parallel_work_amount = [&](const std::shared_ptr<const ov::Node>& n, const ov::PartialShape& shape) {
+        ov::Dimension parallel_work_amount = 1;
+        for (auto it = shape.rbegin() + 2; it != shape.rend(); ++it)
+            parallel_work_amount *= *it;
+        // In dynamic scenario, it is impossible to decide at compilation stage whether parrallel work amount is supported
+        if (parallel_work_amount.is_dynamic())
+            return false;
```

Comment:
If there will be shape `[1,2,?,1024]`, `parallel_work_amount` will be static however `SplitDimensionM` doesn't support dynamic M value and return `false` -> Subgraph won't be tokenized.
Can we extend this check to support tokenization of Subgraph with similar shapes? Probably, we can check for `pshape.is_dynamic()` even.
What do you think?
And need to add the comment with explanation

Reply:
Thanks for the correction, the code is changed, the comment is added